03-objects-and-composition/exercise/06.storeCatalogue.js

```diff
@@ -0,0 +1,16 @@
+function solve(arr) {
+
+    arr.sort((a, b) => a.localeCompare(b));
+
+    let currentLetter;
+
+    arr.forEach(el => {
+        if (el[0][0] != currentLetter) {
+            currentLetter = el[0][0];
+            console.log(currentLetter);
+        }
+
+        let [product, price] = el.split(' : ');
+        console.log(`  ${product}: ${price}`);
+    });
```

Comment:
let sortProducts = Array.from(catalogue.entries()).sort((a, b) => a[0].localeCompare(b[0]));

  for (let [product, price] of sortProducts) {
    getFirstLetter(product);
    if(currentLetter === product[0]) {
      console.log(`  ${product}: ${price}`);
    }
    
  }

  function getFirstLetter(item) {
    if (item[0] !== currentLetter) {
      currentLetter = item[0];
      console.log(currentLetter);
      return currentLetter
    }
  }